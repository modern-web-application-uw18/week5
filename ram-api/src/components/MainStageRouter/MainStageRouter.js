import React, { Component } from 'react';
import { Route, Switch, Redirect } from 'react-router-dom';
import './MainStageRouter.css';
import Home from '../Home/Home';
import Detail from '../Detail/Detail';
import Favorites from '../Favorites/Favorites'
import PropTypes from 'prop-types';

class MainStageRouter extends Component {
    render() {
        return (
            <div className="MainStageRouter">
                <Switch>
                    <Route exact path="/" render={() => {
                        return (<Home searchBoxValue={this.props.searchBoxValue} characters={this.props.characters} />)
                    }} />

                    <Route exact path="/detail/:id" render={() => {
                        return (<Detail favAddRemoveHandler={this.props.favAddRemoveHandler} />)
                    }} />

                    <Route exact path="/favs" render={() => {
                        return (<Favorites characters={this.props.characters} />)
                    }} />
 


                    <Redirect exact from="/detail" to="/" />
                </Switch>
            </div>
        )
    }
}

MainStageRouter.propTypes = {
    searchBoxValue: PropTypes.string,
    favAddRemoveHandler: PropTypes.func,
    characters: PropTypes.array
}

export default MainStageRouter;
