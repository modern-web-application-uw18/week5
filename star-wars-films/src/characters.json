
{
    "characters": {
      "character": [
        {
            "personID": "Luke Skywalker",
            "thumbnail": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQfx6bpsrceI60IE9et0JVgRC8V6L_BCxKJ892NYBUd69KrnKmR",
            "image": "https://i.pinimg.com/originals/8a/27/47/8a2747689d71083d1c766ffff9f37cc1.jpg"
        },
        {
            "personID": "C-3PO",
            "thumbnail": "http://www.starwarshelmets.com/2007/ANH_HD_3po07.jpg",
            "image": "http://fiona-clarke.com/painting/C3PO~0.jpg"
        },
        {
            "personID": "R2-D2",
            "thumbnail": "https://cdn.thisiswhyimbroke.com/images/life-size-r2d2-figure-640x534.jpg",
            "image": "https://i.pinimg.com/736x/78/c7/43/78c743c8fe20b0e20f0f928ea1470d6e--rd-starwars.jpg"
        },
        {
            "personID": "Darth Vader",
            "thumbnail": "http://digitalspyuk.cdnds.net/15/50/980x490/landscape-1449498579-darth-vader-star-wars.jpg",
            "image": "https://images.fineartamerica.com/images/artworkimages/mediumlarge/1/darth-vader-star-wars-michael-greenaway.jpg"
        },
        {
            "personID": "Leia Organa",
            "thumbnail": "https://vignette.wikia.nocookie.net/starwars/images/8/89/Leia_endorpromo02.jpg/revision/latest/scale-to-width-down/500?cb=20080506155343",
            "image": "http://dtlv.com/files/2015/12/jska_priebe_leia_organa_solo_WEB.jpg" 
        },
        {
            "personID": "Owen Lars",
            "thumbnail": "https://vignette.wikia.nocookie.net/starwars/images/9/91/OwenLarsHS-SWE.jpg/revision/latest?cb=20120428164235",
            "image": "https://1.bp.blogspot.com/-qT_LG7Cycvs/ViQtGwhSX3I/AAAAAAAAERE/cl1CsN5ku1Y/s1600/PhilBrown_UncleOwen.jpg" 
        },
        {
            "personID": "Beru Whitesun lars",
            "thumbnail": "https://vignette.wikia.nocookie.net/headhuntersholosuite/images/0/0c/Beru_Lars_002.jpg/revision/latest?cb=20100625172329",
            "image": "https://i.pinimg.com/originals/2b/bb/71/2bbb712405c574c6ce78730e00464a8e.jpg" 
        },
        {
            "personID": "R5-D4",
            "thumbnail": "https://i.pinimg.com/originals/18/22/af/1822af374fe6fd30facbdf47a3b105e2.jpg",
            "image": "https://pre00.deviantart.net/a196/th/pre/i/2016/076/0/9/r5_d4_by_phraggle-d9vgveq.jpg" 
        },
        {
            "personID": "Biggs Darklighter",
            "thumbnail": "https://vignette.wikia.nocookie.net/starwars/images/0/00/BiggsHS-ANH.png/revision/latest?cb=20130305010406",
            "image": "https://vignette.wikia.nocookie.net/starwars/images/2/20/SWEMP8-FC.jpg/revision/latest?cb=20071108175644" 
        },
        {
            "personID": "Obi-Wan Kenobi",
            "thumbnail": "https://zombiesruineverything.files.wordpress.com/2013/11/obs.jpg",
            "image": "https://ixxidesign.azureedge.net/media/1045227/09-Obi-Wan-Kenobi.jpg?width=498" 
        }
      ]
    }
  }
