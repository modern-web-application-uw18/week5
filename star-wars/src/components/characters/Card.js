import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import './Card.css';

class Card extends Component {

  render() {
    const url = `characters/${this.props.data.url.split('/')[5]}`;

    return (
      <div>
        <Link className='character-list' to={url}>{this.props.data.name}</Link>
      </div>
    );
  }
}

Card.propTypes = {
  data: PropTypes.shape({
      name: PropTypes.string,
      birth_year: PropTypes.string
  })
};

export default Card;
