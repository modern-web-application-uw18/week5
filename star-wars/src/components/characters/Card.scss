.character-list {
  text-align: center;
  display: block;
  margin-bottom: 1em;
}
