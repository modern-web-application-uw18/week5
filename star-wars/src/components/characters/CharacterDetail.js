import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import './CharacterDetail.css';
import Spinner from '../widgets/spinner/Spinner';

class CharacterDetail extends Component {
  constructor(props) {
    super(props);
    this.state = {
      data: [], loading: true, characterId: parseInt(this.props.match.params.characterId, 10)
    };
  }

  nextCharacter = () => {
    this.setState((prevState, props) => {
      return { characterId: prevState.characterId + 1, loading: true };
    }, this.updateApiState);
  }

  previousCharacter = () => {
    this.setState((prevState, props) => {
      return { characterId: prevState.characterId - 1, loading: true };
    }, this.updateApiState);
  }

  updateApiState = () => {
    fetch(`https://swapi.co/api/people/${this.state.characterId}`)

    .then(response => response.json())

    .then(data => {
      this.setState((prevState, props) => {
        return { data: data, loading: false};
      });
    })

    .then(() => console.log(this.state.data))
    .catch(error => console.log(error));
  }

  componentDidMount() {
    this.updateApiState();
  }

  render() {
    return (
      <div>
        {this.state.loading &&
          <Spinner />
        }

        {!this.state.loading &&
          <article className='character-detail'>
            <h1 className='character-detail__header'>
              {this.state.data.name}
            </h1>
            <div className='character-detail__details'>
              <div className='character-detail__spec'>Birth year:</div>
              <div className='character-detail__data'>{this.state.data.birth_year}</div>
            </div>
            <div className='character-detail__details'>
              <div className='character-detail__spec'>Height:</div>
              <div className='character-detail__data'>{this.state.data.height} centimeters</div>
            </div>
            <div className='character-detail__details'>
              <div className='character-detail__spec'>Mass:</div>
              <div className='character-detail__data'>{this.state.data.mass} kilograms</div>
            </div>

            <div className='paging-actions'>
              <Link className='btn' to={`/characters/${this.state.characterId - 1}`} onClick={this.previousCharacter}>Previous</Link>
              <Link className='btn' to={`/characters/${this.state.characterId + 1}`} onClick={this.nextCharacter}>Next</Link>
            </div>
          </article>
        }
      </div>
    );
  }
}

CharacterDetail.propTypes = {
  data: PropTypes.shape({
      name: PropTypes.string,
      birth_year: PropTypes.number,
      height: PropTypes.number,
      mass: PropTypes.number
  })
};

export default CharacterDetail
