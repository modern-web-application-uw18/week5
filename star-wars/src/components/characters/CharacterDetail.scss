.character-detail {
  margin-top: 40px;
}

.character-detail__header {
  font-size: 1.3em;
  text-align: center;
  font-family: 'Righteous', cursive;
  margin-bottom: 50px;
}

.character-detail__details {
  display: flex;
  font-size: 1em;
  justify-content: center;
}

.character-detail__data {
  margin-left: 20px;
}

.character-detail__spec {
  position: relative;
  font-weight: bold;
  margin-bottom: 1em;
}
