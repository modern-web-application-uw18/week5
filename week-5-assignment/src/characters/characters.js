import React, {Component} from 'react';
import { Link } from 'react-router-dom';
import '../App.css'
import 'bootstrap/dist/css/bootstrap.css';

class Characters extends Component {
    constructor(props) {
      super(props);
      this.state = {
        results: {},
        page: 1,
    }
}
    componentWillMount() {
      this.getCharacters();
    }
    
    getCharacters = () => {
      fetch(`https://swapi.co/api/people/?page=${this.state.page}`)
      .then(response => response.json())
      .then(data => {
        this.setState(() => {
          return {
            results: data
          };
        });
      });
    }
    
    next = () => {
      this.setState((prevState) => {
        return { page: prevState.page + 1 };
      }, this.getCharacters);
    }
    
    previous = () => {
      this.setState((prevState) => {
        return { page: prevState.page - 1 };
      }, this.getCharacters);
    }

    
    render() {
      return (
        <div>
            <div className="list-group"> 
          {this.state.results.results && this.state.results.results.map((person, idx) => <li className="list-group-item list-group-item-action" onClick={this.getId} index={idx} key={idx}>
            <Link to={`character/${person.name}`}>{person.name}</Link>
          </li>)}
          </div>
          <button className="btn btn-primary" onClick={this.previous}>Previous</button>
          <button className="btn btn-danger"onClick={this.next}>Next</button>
          </div>
       );
    }
  }

  export default Characters;