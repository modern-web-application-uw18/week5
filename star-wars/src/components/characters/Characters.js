import React, { Component } from 'react';
import Spinner from '../widgets/spinner/Spinner';
import Button from '../widgets/button/Button';
import Card from './Card';
import './Characters.css';

class Characters extends Component {

  constructor(props) {
    super(props);
    this.state = {
      data: [], loading: true, paging: 1
    };
  }

  nextCharacterResult = () => {
    this.setState((prevState, props) => {
      return { paging: prevState.paging + 1, loading: true };
    }, this.updateApiState);
  }

  previousCharacterResult = () => {
    this.setState((prevState, props) => {
      return { paging: prevState.paging - 1, loading: true };
    }, this.updateApiState);
  }

  updateApiState = () => {
    fetch(`https://swapi.co/api/people/?page=${this.state.paging}`)
    .then(response => {
      if (response.ok) {
        return response.json()
      } else {
        return Promise.reject('something went wrong!')
       }
    })
    .then(data => {
      this.setState((prevState, props) => {
        return { data: data.results, loading: false};
      });
    })
    .then(() => console.log(this.state.data))
    .catch(error => console.log(error));
  }

  componentDidMount() {
    this.updateApiState();
  }

  render() {

    return (
      <div className='parent'>

        {this.state.loading &&
          <Spinner />
        }

        {!this.state.loading && this.state.data.map (
          (data, index) =>
            <Card data={data} key={index} paging={this.state.paging}/>
          )
        }

        <div className='paging-actions'>
          {!this.state.loading && this.state.paging > 1 &&
           <Button styles='' onClick={this.previousCharacterResult} text='Previous Characters' />
          }

          {!this.state.loading && this.state.paging < 9 &&
           <Button styles='' onClick={this.nextCharacterResult} text='Next Characters' />
          }
        </div>

      </div>
    );
  }
}

export default Characters
