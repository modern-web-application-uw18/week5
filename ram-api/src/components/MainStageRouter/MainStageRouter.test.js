import React from 'react';
import { mount, shallow } from 'enzyme';
import MainStageRouter from './MainStageRouter'
import { MemoryRouter } from 'react-router-dom';
import characters from '../../__mocks__/characters.json'

describe('<MainStageRouter />', () => {
    it('Should render MainStageRouter', () => {

        //https://github.com/ReactTraining/react-router/issues/5579
        //avoid memory router generating new keys everytime it runs so that snapshot remains the same
        const wrapper = shallow(
            <MemoryRouter initialEntries={[{ pathname: '/', key: 'testKey' }]}>
                <MainStageRouter characters={characters} />
            </MemoryRouter>
        );


        expect(wrapper).toMatchSnapshot();
    })
})




