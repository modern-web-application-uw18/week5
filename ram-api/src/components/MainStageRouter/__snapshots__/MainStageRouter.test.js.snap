// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainStageRouter /> Should render MainStageRouter 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "canGo": [Function],
      "createHref": [Function],
      "entries": Array [
        Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/",
          "search": "",
        },
      ],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "index": 0,
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "testKey",
        "pathname": "/",
        "search": "",
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <MainStageRouter
    characters={
      Array [
        Object {
          "created": "2017-11-04T18:48:46.250Z",
          "episode": Array [
            "https://rickandmortyapi.com/api/episode/1",
            "https://rickandmortyapi.com/api/episode/2",
            "https://rickandmortyapi.com/api/episode/3",
            "https://rickandmortyapi.com/api/episode/4",
            "https://rickandmortyapi.com/api/episode/5",
            "https://rickandmortyapi.com/api/episode/6",
            "https://rickandmortyapi.com/api/episode/7",
            "https://rickandmortyapi.com/api/episode/8",
            "https://rickandmortyapi.com/api/episode/9",
            "https://rickandmortyapi.com/api/episode/10",
            "https://rickandmortyapi.com/api/episode/11",
            "https://rickandmortyapi.com/api/episode/12",
            "https://rickandmortyapi.com/api/episode/13",
            "https://rickandmortyapi.com/api/episode/14",
            "https://rickandmortyapi.com/api/episode/15",
            "https://rickandmortyapi.com/api/episode/16",
            "https://rickandmortyapi.com/api/episode/17",
            "https://rickandmortyapi.com/api/episode/18",
            "https://rickandmortyapi.com/api/episode/19",
            "https://rickandmortyapi.com/api/episode/20",
            "https://rickandmortyapi.com/api/episode/21",
            "https://rickandmortyapi.com/api/episode/22",
            "https://rickandmortyapi.com/api/episode/23",
            "https://rickandmortyapi.com/api/episode/24",
            "https://rickandmortyapi.com/api/episode/25",
            "https://rickandmortyapi.com/api/episode/26",
            "https://rickandmortyapi.com/api/episode/27",
            "https://rickandmortyapi.com/api/episode/28",
            "https://rickandmortyapi.com/api/episode/29",
            "https://rickandmortyapi.com/api/episode/30",
            "https://rickandmortyapi.com/api/episode/31",
          ],
          "gender": "Male",
          "id": 1,
          "image": "https://rickandmortyapi.com/api/character/avatar/1.jpeg",
          "location": Object {
            "name": "Earth (Replacement Dimension)",
            "url": "https://rickandmortyapi.com/api/location/20",
          },
          "name": "Rick Sanchez",
          "origin": Object {
            "name": "Earth (C-137)",
            "url": "https://rickandmortyapi.com/api/location/1",
          },
          "species": "Human",
          "status": "Alive",
          "type": "",
          "url": "https://rickandmortyapi.com/api/character/1",
        },
        Object {
          "created": "2017-11-04T18:50:21.651Z",
          "episode": Array [
            "https://rickandmortyapi.com/api/episode/1",
            "https://rickandmortyapi.com/api/episode/2",
            "https://rickandmortyapi.com/api/episode/3",
            "https://rickandmortyapi.com/api/episode/4",
            "https://rickandmortyapi.com/api/episode/5",
            "https://rickandmortyapi.com/api/episode/6",
            "https://rickandmortyapi.com/api/episode/7",
            "https://rickandmortyapi.com/api/episode/8",
            "https://rickandmortyapi.com/api/episode/9",
            "https://rickandmortyapi.com/api/episode/10",
            "https://rickandmortyapi.com/api/episode/11",
            "https://rickandmortyapi.com/api/episode/12",
            "https://rickandmortyapi.com/api/episode/13",
            "https://rickandmortyapi.com/api/episode/14",
            "https://rickandmortyapi.com/api/episode/15",
            "https://rickandmortyapi.com/api/episode/16",
            "https://rickandmortyapi.com/api/episode/17",
            "https://rickandmortyapi.com/api/episode/18",
            "https://rickandmortyapi.com/api/episode/19",
            "https://rickandmortyapi.com/api/episode/20",
            "https://rickandmortyapi.com/api/episode/21",
            "https://rickandmortyapi.com/api/episode/22",
            "https://rickandmortyapi.com/api/episode/23",
            "https://rickandmortyapi.com/api/episode/24",
            "https://rickandmortyapi.com/api/episode/25",
            "https://rickandmortyapi.com/api/episode/26",
            "https://rickandmortyapi.com/api/episode/27",
            "https://rickandmortyapi.com/api/episode/28",
            "https://rickandmortyapi.com/api/episode/29",
            "https://rickandmortyapi.com/api/episode/30",
            "https://rickandmortyapi.com/api/episode/31",
          ],
          "gender": "Male",
          "id": 2,
          "image": "https://rickandmortyapi.com/api/character/avatar/2.jpeg",
          "location": Object {
            "name": "Earth (Replacement Dimension)",
            "url": "https://rickandmortyapi.com/api/location/20",
          },
          "name": "Morty Smith",
          "origin": Object {
            "name": "Earth (C-137)",
            "url": "https://rickandmortyapi.com/api/location/1",
          },
          "species": "Human",
          "status": "Alive",
          "type": "",
          "url": "https://rickandmortyapi.com/api/character/2",
        },
        Object {
          "created": "2017-11-04T19:09:56.428Z",
          "episode": Array [
            "https://rickandmortyapi.com/api/episode/6",
            "https://rickandmortyapi.com/api/episode/7",
            "https://rickandmortyapi.com/api/episode/8",
            "https://rickandmortyapi.com/api/episode/9",
            "https://rickandmortyapi.com/api/episode/10",
            "https://rickandmortyapi.com/api/episode/11",
            "https://rickandmortyapi.com/api/episode/12",
            "https://rickandmortyapi.com/api/episode/14",
            "https://rickandmortyapi.com/api/episode/15",
            "https://rickandmortyapi.com/api/episode/16",
            "https://rickandmortyapi.com/api/episode/17",
            "https://rickandmortyapi.com/api/episode/18",
            "https://rickandmortyapi.com/api/episode/19",
            "https://rickandmortyapi.com/api/episode/20",
            "https://rickandmortyapi.com/api/episode/21",
            "https://rickandmortyapi.com/api/episode/22",
            "https://rickandmortyapi.com/api/episode/23",
            "https://rickandmortyapi.com/api/episode/24",
            "https://rickandmortyapi.com/api/episode/25",
            "https://rickandmortyapi.com/api/episode/26",
            "https://rickandmortyapi.com/api/episode/27",
            "https://rickandmortyapi.com/api/episode/29",
            "https://rickandmortyapi.com/api/episode/30",
            "https://rickandmortyapi.com/api/episode/31",
          ],
          "gender": "Female",
          "id": 3,
          "image": "https://rickandmortyapi.com/api/character/avatar/3.jpeg",
          "location": Object {
            "name": "Earth (Replacement Dimension)",
            "url": "https://rickandmortyapi.com/api/location/20",
          },
          "name": "Summer Smith",
          "origin": Object {
            "name": "Earth (Replacement Dimension)",
            "url": "https://rickandmortyapi.com/api/location/20",
          },
          "species": "Human",
          "status": "Alive",
          "type": "",
          "url": "https://rickandmortyapi.com/api/character/3",
        },
        Object {
          "created": "2017-11-04T19:22:43.665Z",
          "episode": Array [
            "https://rickandmortyapi.com/api/episode/6",
            "https://rickandmortyapi.com/api/episode/7",
            "https://rickandmortyapi.com/api/episode/8",
            "https://rickandmortyapi.com/api/episode/9",
            "https://rickandmortyapi.com/api/episode/10",
            "https://rickandmortyapi.com/api/episode/11",
            "https://rickandmortyapi.com/api/episode/12",
            "https://rickandmortyapi.com/api/episode/14",
            "https://rickandmortyapi.com/api/episode/15",
            "https://rickandmortyapi.com/api/episode/16",
            "https://rickandmortyapi.com/api/episode/18",
            "https://rickandmortyapi.com/api/episode/19",
            "https://rickandmortyapi.com/api/episode/20",
            "https://rickandmortyapi.com/api/episode/21",
            "https://rickandmortyapi.com/api/episode/22",
            "https://rickandmortyapi.com/api/episode/23",
            "https://rickandmortyapi.com/api/episode/24",
            "https://rickandmortyapi.com/api/episode/25",
            "https://rickandmortyapi.com/api/episode/26",
            "https://rickandmortyapi.com/api/episode/27",
            "https://rickandmortyapi.com/api/episode/28",
            "https://rickandmortyapi.com/api/episode/29",
            "https://rickandmortyapi.com/api/episode/30",
            "https://rickandmortyapi.com/api/episode/31",
          ],
          "gender": "Female",
          "id": 4,
          "image": "https://rickandmortyapi.com/api/character/avatar/4.jpeg",
          "location": Object {
            "name": "Earth (Replacement Dimension)",
            "url": "https://rickandmortyapi.com/api/location/20",
          },
          "name": "Beth Smith",
          "origin": Object {
            "name": "Earth (Replacement Dimension)",
            "url": "https://rickandmortyapi.com/api/location/20",
          },
          "species": "Human",
          "status": "Alive",
          "type": "",
          "url": "https://rickandmortyapi.com/api/character/4",
        },
        Object {
          "created": "2017-11-04T19:26:56.301Z",
          "episode": Array [
            "https://rickandmortyapi.com/api/episode/6",
            "https://rickandmortyapi.com/api/episode/7",
            "https://rickandmortyapi.com/api/episode/8",
            "https://rickandmortyapi.com/api/episode/9",
            "https://rickandmortyapi.com/api/episode/10",
            "https://rickandmortyapi.com/api/episode/11",
            "https://rickandmortyapi.com/api/episode/12",
            "https://rickandmortyapi.com/api/episode/13",
            "https://rickandmortyapi.com/api/episode/14",
            "https://rickandmortyapi.com/api/episode/15",
            "https://rickandmortyapi.com/api/episode/16",
            "https://rickandmortyapi.com/api/episode/18",
            "https://rickandmortyapi.com/api/episode/19",
            "https://rickandmortyapi.com/api/episode/20",
            "https://rickandmortyapi.com/api/episode/21",
            "https://rickandmortyapi.com/api/episode/22",
            "https://rickandmortyapi.com/api/episode/23",
            "https://rickandmortyapi.com/api/episode/26",
            "https://rickandmortyapi.com/api/episode/29",
            "https://rickandmortyapi.com/api/episode/30",
            "https://rickandmortyapi.com/api/episode/31",
          ],
          "gender": "Male",
          "id": 5,
          "image": "https://rickandmortyapi.com/api/character/avatar/5.jpeg",
          "location": Object {
            "name": "Earth (Replacement Dimension)",
            "url": "https://rickandmortyapi.com/api/location/20",
          },
          "name": "Jerry Smith",
          "origin": Object {
            "name": "Earth (Replacement Dimension)",
            "url": "https://rickandmortyapi.com/api/location/20",
          },
          "species": "Human",
          "status": "Alive",
          "type": "",
          "url": "https://rickandmortyapi.com/api/character/5",
        },
        Object {
          "created": "2017-11-04T19:50:28.250Z",
          "episode": Array [
            "https://rickandmortyapi.com/api/episode/27",
          ],
          "gender": "Female",
          "id": 6,
          "image": "https://rickandmortyapi.com/api/character/avatar/6.jpeg",
          "location": Object {
            "name": "Abadango",
            "url": "https://rickandmortyapi.com/api/location/2",
          },
          "name": "Abadango Cluster Princess",
          "origin": Object {
            "name": "Abadango",
            "url": "https://rickandmortyapi.com/api/location/2",
          },
          "species": "Alien",
          "status": "Alive",
          "type": "",
          "url": "https://rickandmortyapi.com/api/character/6",
        },
        Object {
          "created": "2017-11-04T19:59:20.523Z",
          "episode": Array [
            "https://rickandmortyapi.com/api/episode/10",
            "https://rickandmortyapi.com/api/episode/11",
          ],
          "gender": "Male",
          "id": 7,
          "image": "https://rickandmortyapi.com/api/character/avatar/7.jpeg",
          "location": Object {
            "name": "Testicle Monster Dimension",
            "url": "https://rickandmortyapi.com/api/location/21",
          },
          "name": "Abradolf Lincler",
          "origin": Object {
            "name": "Earth (Replacement Dimension)",
            "url": "https://rickandmortyapi.com/api/location/20",
          },
          "species": "Human",
          "status": "unknown",
          "type": "Genetic experiment",
          "url": "https://rickandmortyapi.com/api/character/7",
        },
        Object {
          "created": "2017-11-04T20:03:34.737Z",
          "episode": Array [
            "https://rickandmortyapi.com/api/episode/28",
          ],
          "gender": "Male",
          "id": 8,
          "image": "https://rickandmortyapi.com/api/character/avatar/8.jpeg",
          "location": Object {
            "name": "Citadel of Ricks",
            "url": "https://rickandmortyapi.com/api/location/3",
          },
          "name": "Adjudicator Rick",
          "origin": Object {
            "name": "unknown",
            "url": "",
          },
          "species": "Human",
          "status": "Dead",
          "type": "",
          "url": "https://rickandmortyapi.com/api/character/8",
        },
        Object {
          "created": "2017-11-04T20:06:54.976Z",
          "episode": Array [
            "https://rickandmortyapi.com/api/episode/24",
          ],
          "gender": "Male",
          "id": 9,
          "image": "https://rickandmortyapi.com/api/character/avatar/9.jpeg",
          "location": Object {
            "name": "Earth (Replacement Dimension)",
            "url": "https://rickandmortyapi.com/api/location/20",
          },
          "name": "Agency Director",
          "origin": Object {
            "name": "Earth (Replacement Dimension)",
            "url": "https://rickandmortyapi.com/api/location/20",
          },
          "species": "Human",
          "status": "Dead",
          "type": "",
          "url": "https://rickandmortyapi.com/api/character/9",
        },
        Object {
          "created": "2017-11-04T20:19:09.017Z",
          "episode": Array [
            "https://rickandmortyapi.com/api/episode/25",
          ],
          "gender": "Male",
          "id": 10,
          "image": "https://rickandmortyapi.com/api/character/avatar/10.jpeg",
          "location": Object {
            "name": "Worldender's lair",
            "url": "https://rickandmortyapi.com/api/location/4",
          },
          "name": "Alan Rails",
          "origin": Object {
            "name": "unknown",
            "url": "",
          },
          "species": "Human",
          "status": "Dead",
          "type": "Superhuman (Ghost trains summoner)",
          "url": "https://rickandmortyapi.com/api/character/10",
        },
        Object {
          "created": "2017-11-04T20:20:20.965Z",
          "episode": Array [
            "https://rickandmortyapi.com/api/episode/12",
          ],
          "gender": "Male",
          "id": 11,
          "image": "https://rickandmortyapi.com/api/character/avatar/11.jpeg",
          "location": Object {
            "name": "Earth (Replacement Dimension)",
            "url": "https://rickandmortyapi.com/api/location/20",
          },
          "name": "Albert Einstein",
          "origin": Object {
            "name": "Earth (C-137)",
            "url": "https://rickandmortyapi.com/api/location/1",
          },
          "species": "Human",
          "status": "Dead",
          "type": "",
          "url": "https://rickandmortyapi.com/api/character/11",
        },
        Object {
          "created": "2017-11-04T20:32:33.144Z",
          "episode": Array [
            "https://rickandmortyapi.com/api/episode/3",
          ],
          "gender": "Male",
          "id": 12,
          "image": "https://rickandmortyapi.com/api/character/avatar/12.jpeg",
          "location": Object {
            "name": "Anatomy Park",
            "url": "https://rickandmortyapi.com/api/location/5",
          },
          "name": "Alexander",
          "origin": Object {
            "name": "Earth (C-137)",
            "url": "https://rickandmortyapi.com/api/location/1",
          },
          "species": "Human",
          "status": "Dead",
          "type": "",
          "url": "https://rickandmortyapi.com/api/character/12",
        },
        Object {
          "created": "2017-11-04T20:33:30.779Z",
          "episode": Array [
            "https://rickandmortyapi.com/api/episode/31",
          ],
          "gender": "unknown",
          "id": 13,
          "image": "https://rickandmortyapi.com/api/character/avatar/13.jpeg",
          "location": Object {
            "name": "Earth (Replacement Dimension)",
            "url": "https://rickandmortyapi.com/api/location/20",
          },
          "name": "Alien Googah",
          "origin": Object {
            "name": "unknown",
            "url": "",
          },
          "species": "Alien",
          "status": "unknown",
          "type": "",
          "url": "https://rickandmortyapi.com/api/character/13",
        },
        Object {
          "created": "2017-11-04T20:51:31.373Z",
          "episode": Array [
            "https://rickandmortyapi.com/api/episode/10",
          ],
          "gender": "Male",
          "id": 14,
          "image": "https://rickandmortyapi.com/api/character/avatar/14.jpeg",
          "location": Object {
            "name": "Citadel of Ricks",
            "url": "https://rickandmortyapi.com/api/location/3",
          },
          "name": "Alien Morty",
          "origin": Object {
            "name": "unknown",
            "url": "",
          },
          "species": "Alien",
          "status": "unknown",
          "type": "",
          "url": "https://rickandmortyapi.com/api/character/14",
        },
        Object {
          "created": "2017-11-04T20:56:13.215Z",
          "episode": Array [
            "https://rickandmortyapi.com/api/episode/10",
          ],
          "gender": "Male",
          "id": 15,
          "image": "https://rickandmortyapi.com/api/character/avatar/15.jpeg",
          "location": Object {
            "name": "Citadel of Ricks",
            "url": "https://rickandmortyapi.com/api/location/3",
          },
          "name": "Alien Rick",
          "origin": Object {
            "name": "unknown",
            "url": "",
          },
          "species": "Alien",
          "status": "unknown",
          "type": "",
          "url": "https://rickandmortyapi.com/api/character/15",
        },
        Object {
          "created": "2017-11-04T21:12:45.235Z",
          "episode": Array [
            "https://rickandmortyapi.com/api/episode/15",
          ],
          "gender": "Male",
          "id": 16,
          "image": "https://rickandmortyapi.com/api/character/avatar/16.jpeg",
          "location": Object {
            "name": "Earth (Replacement Dimension)",
            "url": "https://rickandmortyapi.com/api/location/20",
          },
          "name": "Amish Cyborg",
          "origin": Object {
            "name": "unknown",
            "url": "",
          },
          "species": "Alien",
          "status": "Dead",
          "type": "Parasite, Cyborg",
          "url": "https://rickandmortyapi.com/api/character/16",
        },
        Object {
          "created": "2017-11-04T22:21:24.481Z",
          "episode": Array [
            "https://rickandmortyapi.com/api/episode/3",
          ],
          "gender": "Female",
          "id": 17,
          "image": "https://rickandmortyapi.com/api/character/avatar/17.jpeg",
          "location": Object {
            "name": "Anatomy Park",
            "url": "https://rickandmortyapi.com/api/location/5",
          },
          "name": "Annie",
          "origin": Object {
            "name": "Earth (C-137)",
            "url": "https://rickandmortyapi.com/api/location/1",
          },
          "species": "Human",
          "status": "Alive",
          "type": "",
          "url": "https://rickandmortyapi.com/api/character/17",
        },
        Object {
          "created": "2017-11-04T22:25:29.008Z",
          "episode": Array [
            "https://rickandmortyapi.com/api/episode/10",
            "https://rickandmortyapi.com/api/episode/28",
          ],
          "gender": "Male",
          "id": 18,
          "image": "https://rickandmortyapi.com/api/character/avatar/18.jpeg",
          "location": Object {
            "name": "Citadel of Ricks",
            "url": "https://rickandmortyapi.com/api/location/3",
          },
          "name": "Antenna Morty",
          "origin": Object {
            "name": "unknown",
            "url": "",
          },
          "species": "Human",
          "status": "Alive",
          "type": "Human with antennae",
          "url": "https://rickandmortyapi.com/api/character/18",
        },
        Object {
          "created": "2017-11-04T22:28:13.756Z",
          "episode": Array [
            "https://rickandmortyapi.com/api/episode/10",
          ],
          "gender": "Male",
          "id": 19,
          "image": "https://rickandmortyapi.com/api/character/avatar/19.jpeg",
          "location": Object {
            "name": "unknown",
            "url": "",
          },
          "name": "Antenna Rick",
          "origin": Object {
            "name": "unknown",
            "url": "",
          },
          "species": "Human",
          "status": "unknown",
          "type": "Human with antennae",
          "url": "https://rickandmortyapi.com/api/character/19",
        },
        Object {
          "created": "2017-11-04T22:34:53.659Z",
          "episode": Array [
            "https://rickandmortyapi.com/api/episode/8",
          ],
          "gender": "Male",
          "id": 20,
          "image": "https://rickandmortyapi.com/api/character/avatar/20.jpeg",
          "location": Object {
            "name": "Interdimensional Cable",
            "url": "https://rickandmortyapi.com/api/location/6",
          },
          "name": "Ants in my Eyes Johnson",
          "origin": Object {
            "name": "unknown",
            "url": "",
          },
          "species": "Human",
          "status": "unknown",
          "type": "Human with ants in his eyes",
          "url": "https://rickandmortyapi.com/api/character/20",
        },
      ]
    }
  />
</Router>
`;
